etcd-client/src/lib.rs

```diff
@@ -77,7 +77,13 @@
     clippy::implicit_return, // actually omitting the return keyword is idiomatic Rust code
     clippy::module_name_repetitions, // repeation of module name in a struct name is not big deal
     clippy::multiple_crate_versions, // multi-version dependency crates is not able to fix
-    clippy::panic, // allow debug_assert, panic in production code
+    clippy::panic, // allow debug_assert, panic in production code  
+    clippy::shadow_same, //it’s a common pattern in Rust code
+    clippy::shadow_unrelated,
+    clippy::shadow_reuse,
+    clippy::same_name_method,
+    clippy::separated_literal_suffix,
+    clippy::mod_module_files, // TODO: fix code structure to pass this lint
```

Comment:
Why do we allow these clippy?

Reply:
The reasons are added in the recent commit, which are caused by rust version update and there are some conflict clippy rule(we have to choose one of them) and some are for the generated code.